presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/MapColumnStatisticsBuilder.java

```diff
@@ -45,14 +53,22 @@ public void addMapStatistics(DwrfProto.KeyInfo key, ColumnStatistics columnStati
     {
         requireNonNull(key, "key is null");
         requireNonNull(columnStatistics, "columnStatistics is null");
-        nonNullValueCount += columnStatistics.getNumberOfValues();
```

Comment:
I made a mistake in the initial implementation confusing map row with map value. To fix it I added explicit increaseValueCount.